x-pack/plugins/enterprise_search/public/applications/app_search/app_logic.ts

```diff
@@ -26,5 +33,93 @@ export const AppLogic = kea<MakeLogicType<IAppValues, IAppActions>>({
         initializeAppData: () => true,
       },
     ],
+    account: [
+      {} as IAccount,
+      {
+        initializeAppData: (_, { appSearch: account }) => account,
+        setOnboardingComplete: (state) => ({
+          ...state,
+          onboardingComplete: true,
+        }),
+      },
+    ],
+    configuredLimits: [
+      {} as IConfiguredLimits,
+      {
+        initializeAppData: (_, { configuredLimits }) => configuredLimits.appSearch,
+      },
+    ],
+    ilmEnabled: [
+      false,
+      {
+        initializeAppData: (_, { ilmEnabled }) => !!ilmEnabled,
+      },
+    ],
+  },
+  selectors: {
+    /**
+     * Massages the `role` data we receive from the Enterprise Search
+     * server into a more convenient format for front-end use
+     */
+    myRole: [
```

Comment:
I like this change _a lot_. I'd probably move it to its own file personally

Reply:
I like this change Constance. Do you think it will be a pain to deal with as we migrate consuming code over, or do you think it should have a minimal impact?

Reply:
If I were going to separate it out, I think I would separate out a plain old javascript function that adapts a server role to a front-end role. It could just be a plain old javascript function, which could make it easy to test on it's own AND tighten up this file.

Your call though, I think it's fine inline for now as well.

Reply:
Sweet. Sounds like there's a strong case for moving this to its own separate file/function (definitely doable). 👍 Will do so here!

Reply:
> Do you think it will be a pain to deal with as we migrate consuming code over, or do you think it should have a minimal impact?

The biggest impact is the hook:
```js
const { canViewEngines } = useAbilities();`
```
becomes
```
const { myRole: { canViewEngines } } = useValues(AppLogic);
```

But I think those type of changes are part and parcel of the migration (e.g., `handleAPIError` -> `flashAPIErrors`), and we can either document them in a checklist or they'll eventually become muscle memory.

Reply:
I moved all the myRole logic into its own `utils/role/` file here: de93ff9

Turns out Kea was maybe squashing some type errors it shouldn't have been, so pulling it out to a separate file helped catch those. Awesome!

Also would love y'alls thoughts on me moving role types into its relevant file here: ae252fc

Reply:
All of it looks great, including moving type roles into it's own file.